_includes/markdown/React.md

```diff
@@ -33,6 +33,25 @@ If you already have a Node / NPM / Yarn project that you wish to include React a
 
 Using the CLI tool is ideal if you are creating a full React App with an API backend.
 
+## Components
+When building out components, its beneficial to understand how to construct them in the most appropriate way possible. Certain "types" of components can be written differently which can have big performance benefits on larger scale applications.
+
+### Class Components
+Class Components are written in the ES6 Class syntax. When building a component using a JS Class, you are generally inferring that the component either manages its own `state` or it's `state` is managed by a state management library like Redux. LifeCycle methods are only available in class components. 
+
+Class components are also capabale of handling `props`. Use these components when you need to build "intelligent" React components that are aware of their own `state` as well as the state of their children.
```

Comment:
@dainemawer can we add a basic example of when this is applicable? Like, "If you're building xyz, it might be a good application of Class Components". and something similar for stateless and pure as well

Reply:
Hey @timwright12 I can do that! Will update the PR accordingly!

Reply:
thanks, ping me when it's in there

Reply:
@timwright12 updated those points - for Pure Components, its bit tricky to give an example of a specific component, but I tried to clarify that. Let me know what you think